pkg/apis/pravega/v1alpha1/pravega.go

```diff
@@ -65,83 +65,206 @@ const (
 	DefaultSegmentStoreLimitMemory = "2Gi"
 )
 
-// PravegaSpec defines the configuration of Pravega
-type PravegaSpec struct {
+// ExternalAccess defines the configuration of the external access
+type ExternalAccess struct {
+	// Type specifies the service type to achieve external access.
+	// Options are "LoadBalancer" and "NodePort".
+	// By default, if external access is enabled, it will use "LoadBalancer"
+	Type v1.ServiceType `json:"type,omitempty"`
+
+	// Domain Name to be used for External Access
+	// If empty no dns name is added for
+	DomainName string `json:"domainName,omitempty"`
+
+	// Annotations to be added to the external service
+	Annotations map[string]string `json:"annotations"`
+}
+
+func (e *ExternalAccess) withDefaults() (changed bool) {
+	if e.Type == "" {
+		changed = true
+		e.Type = DefaultServiceType
+	}
+
+	if e.Annotations == nil {
+		changed = true
+		e.Annotations = map[string]string{}
+	}
+
+	return changed
+}
+
+type ControllerSpec struct {
 	// ControllerReplicas defines the number of Controller replicas.
 	// Defaults to 1.
-	ControllerReplicas int32 `json:"controllerReplicas"`
+	Replicas int32 `json:"replicas"`
+
+	// DebugLogging indicates whether or not debug level logging is enabled.
+	// Defaults to false.
+	DebugLogging bool `json:"debugLogging"`
+
+	// Pravega configuration options passed to the Pravega processes
+	// a '-D' is prefixed to the provided values to pass them as java system properties
+	// See the following file for a complete list of options:
+	// https://github.com/pravega/pravega/blob/master/config/config.properties
+	Options map[string]string `json:"options"`
```

Comment:
It would be great to have an example on how to set these very useful `options` (and `jvmOptions`) separately on segmentstore and controller specs.

Reply:
Provided this in PR comments.